MichaelKintscher/qualitative-coding-app
# Groups all sessions together in QSettings

Before this patch, sessions were assumed to be the first level of the QSettings key heirarchy. Now, with the introduction of global-settings, this is no longer the case. With this patch, we group together all sessions into a sessions group, allowing our first level of key heirarchy to be independently isolated.

Testing Steps:
  1. Set the git head to the commit prior to this one to remove its changes. 1a. Perform a git log command in the terminal 2a. Copy the commit ID of the commit just below this commit
    3a. Perform the command: git reset --hard <commit id>
  2. Run the application and clear all sessions, then close the window.
  3. Pull the branch again to get this commit back.
  4. Run the application and create a new session, "session 1"
  5. Enter dummy data into the encoding table and close the window
  6. Repeat steps 4-5, but this time name the session, "session 2"
  7. Verify the dummy data persists in sessions 1 and 2 by loading the sessions through the welcome screen session management page, as well as the session management page accessed from the application toolbar
  8. Delete "session 1" and restart the application to verify that it has been removed
  9. Create a new session, "session 3" and close the window.
  10. Run the application and clear all sessions, verify they have all been removed.
  11. Mac users, verify "Com" is not presented in the session list
  12. Verify that "global-settings" is not presented in the session list

Type: Refactor
